ddl/ddl_api.go

```diff
@@ -4340,14 +4340,71 @@ func (d *ddl) RenameTable(ctx sessionctx.Context, oldIdent, newIdent ast.Ident,
 		SchemaName: newSchema.Name.L,
 		Type:       model.ActionRenameTable,
 		BinlogInfo: &model.HistoryInfo{},
-		Args:       []interface{}{oldSchema.ID, newIdent.Name},
+		Args:       []interface{}{[]int64{oldSchema.ID}, newIdent.Name},
 	}
 
 	err = d.doDDLJob(ctx, job)
 	err = d.callHookOnChanged(err)
 	return errors.Trace(err)
 }
 
+func (d *ddl) RenameTables(ctx sessionctx.Context, oldIdents, newIdents []ast.Ident, isAlterTable bool) error {
+	is := d.GetInfoSchemaWithInterceptor(ctx)
+	tableNames := make([]*model.CIStr, 0, len(oldIdents))
+	oldSchemaIDs := make([]int64, 0, len(oldIdents))
+	tableIDs := make([]int64, 0, len(oldIdents))
+	var newSchema *model.DBInfo
+	var tableID int64
+	for i := 0; i < len(oldIdents); i++ {
+		oldSchema, ok := is.SchemaByName(oldIdents[i].Schema)
+		if !ok {
+			if isAlterTable {
+				return infoschema.ErrTableNotExists.GenWithStackByArgs(oldIdents[i].Schema, oldIdents[i].Name)
+			}
+			if is.TableExists(newIdents[i].Schema, newIdents[i].Name) {
+				return infoschema.ErrTableExists.GenWithStackByArgs(newIdents[i])
+			}
+			return errFileNotFound.GenWithStackByArgs(oldIdents[i].Schema, oldIdents[i].Name)
+		}
+		oldTbl, err := is.TableByName(oldIdents[i].Schema, oldIdents[i].Name)
+		if err != nil {
+			if isAlterTable {
+				return infoschema.ErrTableNotExists.GenWithStackByArgs(oldIdents[i].Schema, oldIdents[i].Name)
+			}
+			return errFileNotFound.GenWithStackByArgs(oldIdents[i].Schema, oldIdents[i].Name)
+		}
+		if isAlterTable && newIdents[i].Schema.L == oldIdents[i].Schema.L && newIdents[i].Name.L == oldIdents[i].Name.L {
+			//oldIdent is equal to newIdent, do nothing
+			return nil
+		}
+		newSchema, ok = is.SchemaByName(newIdents[i].Schema)
+		if !ok {
+			return ErrErrorOnRename.GenWithStackByArgs(
+				fmt.Sprintf("%s.%s", oldIdents[i].Schema, oldIdents[i].Name),
+				fmt.Sprintf("%s.%s", newIdents[i].Schema, newIdents[i].Name),
+				168,
+				fmt.Sprintf("Database `%s` doesn't exist", newIdents[i].Schema))
```

Comment:
Could we extract line4359-line4386 into a function? Then, both `RenameTable` and `RenameTables` can use this function.